src/cpp/profiling/endoscope_frontend.cc

```diff
@@ -0,0 +1,147 @@
+#include "src/cpp/profiling/endoscope_frontend.h"
+
+#ifdef GRPC_ENDOSCOPE_PROFILER
```

Comment:
Yes. Without it, the backend doesn't exist, and data structures in frontend are not defined.